src/python/pants/task/simple_codegen_task.py

```diff
@@ -359,6 +359,7 @@ def _inject_synthetic_target(self, vt, sources):
     synthetic_target = self.context.add_new_target(
       address=self._get_synthetic_address(target, synthetic_target_dir),
       target_type=synthetic_target_type,
+      type_alias=synthetic_target_type.alias(),
```

Comment:
This is assuming the target type has defined an `alias` classmethod, which many do, but not all. Could we add a check with `hasattr` here?

Reply:
Alternatively, define this as an explicitly overridable method similar to: https://github.com/pantsbuild/pants/blob/805d36e4df797f2405604feca10df9148f74244d/src/python/pants/task/simple_codegen_task.py#L120-L134

Reply:
^In support of the above.